static/js/components/dashboard/courses/StatusMessages.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+import React from 'react';
+import R from 'ramda';
+import moment from 'moment';
+
+import type { Coupon } from '../../../flow/couponTypes';
+import type { Course, CourseRun } from '../../../flow/programTypes';
+import { couponMessageText } from '../CouponMessage';
+import { makeCouponReason } from '../../../lib/coupon';
+import {
+  COUPON_CONTENT_TYPE_COURSE,
+  COURSE_ACTION_PAY,
+  COURSE_ACTION_REENROLL,
+} from '../../../constants';
+import { S } from '../../../lib/sanctuary';
+import {
+  hasPaidForAnyCourseRun,
+  courseUpcomingOrCurrent,
+  hasPassedCourseRun,
+  hasFailedCourseRun,
+  futureEnrollableRun,
+} from './util';
+import { hasPassingExamGrade } from '../../../lib/grades';
+
+type Message = {
+  message: string|React$Element<*>,
+  action?: React$Element<*>
+};
+
+export const formatAction = (message: Message) => (
+  <div className="second-col">
+    { message.action }
+  </div>
+);
+
+export const formatMessage = (message: Message, index?: number) => (
+  <div className="status-message cols" key={index}>
+    <div className="message first-col">
+      { message.message }
+    </div>
+    { message.action ? formatAction(message) : null }
+  </div>
+);
+
+const formatDate = date => (
+  moment(date).format('MM/DD/YYYY')
```

Comment:
Can you extract this and move it to `constants.js` near the other date format strings?